character-details.mjs

```diff
@@ -38,41 +38,45 @@ character-details-page .character-details p span {
 `;
 document.body.appendChild(styles);
 
-export default Component.extend({
-  tag: "character-details-page",
+export default class CharacterDetailsPage extends StacheElement {
+	static view = `
+		<div class="breadcrumbs">
+			<div>
+				<a href="{{ routeUrl(page="list" query=query)}}" class="search">&lt; Characters</a>
+			</div>
+		</div>
 
-  view: `
-    <div class="breadcrumbs">
-      <div>
-        <a href="{{ routeUrl(page="list" query=query)}}" class="search">&lt; Characters</a>
-      </div>
-    </div>
+		{{# if(characterPromise.isPending) }}
+			<div class="loading">Loading...</div>
+		{{/ if }}
 
-    {{# if(characterPromise.isPending) }}
-      <div class="loading">Loading...</div>
-    {{/ if }}
+		{{# if(characterPromise.isResolved) }}
+			<div class="character-details">
+				{{# with(characterPromise.value) }}
+					<h2>{{name}}</h2>
+					<img src="{{image}}" alt="{{name}}"/>
+					<p><span>Status:</span> {{status}}</p>
+					<p><span>Species:</span> {{species}}</p>
+					<p><span>Location:</span> {{location.name}}</p>
+					<p><span>Type:</span> {{# if(type) }}{{type}}{{ else }}Unknown{{/ if }}</p>
+				{{/ with }}
+			</div>
+		{{/ if }}
+	`;
 
-    {{# if(characterPromise.isResolved) }}
-      <div class="character-details">
-        {{# with(characterPromise.value) }}
-          <h2>{{name}}</h2>
-          <img src="{{image}}" alt="{{name}}"/>
-          <p><span>Status:</span> {{status}}</p>
-          <p><span>Species:</span> {{species}}</p>
-          <p><span>Location:</span> {{location.name}}</p>
-          <p><span>Type:</span> {{# if(type) }}{{type}}{{ else }}Unknown{{/ if }}</p>
-        {{/ with }}
-      </div>
-    {{/ if }}
-  `,
+	static props = {
+		query: type.maybeConvert(String),
+		id: type.maybeConvert(Number),
 
-  ViewModel: {
-    query: "string",
-    id: "number",
+		get characterPromise() {
+			return (
+				this.id &&
```

Comment:
@phillipskevin I had to add this conditional here, unlike the can@5 version, the new one re-renders `character-details-page` when id is removed from the url (set to undefined). I actually get stache warning of properties not found in the scope when navigating from the details page to the list. We might want to look into this and make sure it's fine?

Reply:
scratch that, I was using `can@5`.... I cannot replicate in `can@pre`.